packages/vm/test/t8n/t8ntool.ts

```diff
@@ -164,43 +164,78 @@ export class TransitionTool {
 
     let index = 0
 
-    this.vm.events.on('afterTx', (event) => {
-      this.afterTx(event, index, builder)
-    })
+    let trace: any[] = []
+    // Tracing
+    if (args.trace === true) {
+      this.vm.evm.events?.on('step', (e) => {
+        let hexStack = []
+        hexStack = e.stack.map((item: bigint) => {
+          return '0x' + item.toString(16)
+        })
+        const opTrace = {
+          pc: e.pc,
+          op: e.opcode.name,
+          gas: bigIntToHex(e.gasLeft),
+          gasCost: bigIntToHex(BigInt(e.opcode.fee)),
+          stack: hexStack,
+          depth: e.depth,
+          opName: e.opcode.name,
+        }
+        trace.push(JSON.stringify(opTrace))
+      })
+      this.vm.events.on('afterTx', async (event) => {
```

Comment:
There is something we need to take into account, and that is that there is also EVM execution outside of the tx context (system contracts). Therefore the `trace` array will fill up with data if EVM executes code before a tx is ran (and this is the case I think for EIP 2935 and maybe others). So, we should clear the `trace` array `beforeTx` (and setup the context such that all is clear to get logged for the current tx in `afterTx`. We can even skip the clearing there because this happens in the `beforeTx` if we changed it)